pkg/repository/git.go

```diff
@@ -120,6 +123,51 @@ func (r gitRepoController) getSHA() (string, error) {
 	return head.Hash().String(), nil
 }
 
+func (r gitRepoController) getTreeSHA(context string) (string, error) {
+	repo, err := r.repoGetter.get(r.path)
+	if err != nil {
+		return "", fmt.Errorf("failed to get repository: %w", err)
+	}
+
+	ref, err := repo.Head()
+	if err != nil {
+		return "", fmt.Errorf("failed to get HEAD from repo: %w", err)
+	}
+
+	commit, err := repo.CommitObject(ref.Hash())
+	if err != nil {
+		return "", fmt.Errorf("failed to get commit object from reference: %w", err)
+	}
+	tree, err := commit.Tree()
+	if err != nil {
+		return "", fmt.Errorf("failed to get tree from commit: %w", err)
+	}
+
+	if context == "." {
+		return tree.Hash.String(), nil
+	}
+
+	treeDir := getTreeDirFromRelPath(context)
+
+	svcEntry, err := tree.FindEntry(treeDir)
+	if err != nil {
+		return "", fmt.Errorf("failed to find an entry in tree: %w", err)
+	}
+
+	return svcEntry.Hash.String(), nil
+}
+
+func getTreeDirFromRelPath(relPath string) string {
+
+	wdCtrl := filesystem.NewOsWorkingDirectoryCtrl()
+	wd, err := wdCtrl.Get()
+	if err != nil {
```

Comment:
Same as https://github.com/okteto/okteto/pull/4023#discussion_r1371621771

It feels odd to have logs inside the function, it could up to the calling function to decide wheather to ignore the error and log. WDYT?

Reply:
acd7ef13f7f00dedac81e167c82100f3e202eb55